File: lua/vimade/config_helpers/tint.lua
local M = {}
local COLOR_UTIL = require('vimade.util.color')
local GLOBALS

M.__init = function (globals)
  GLOBALS = globals
end

M.MIX = 'MIX' -- default
M.REPLACE = 'REPLACE' 

M.DEFAULT = function (win)
  if GLOBALS.basebg then
    return {
      -- basebg was previously used as a semi-tint mechanism used in place of the Normalbg
      -- this doesn't exactly reproduce the same colors but gets close
      fg = {
        rgb = COLOR_UTIL.toRgb(GLOBALS.basebg),
        intensity = 0.5,
        type = M.MIX,
      }
    }
  else
    return nil
  end
end

return M

File: lib/vimade/v2/config_helpers/tint.py
import sys
M = sys.modules[__name__]

from vimade.v2.util import color as COLOR_UTIL
GLOBALS = None

def __init(globals):
  global GLOBALS
  GLOBALS = globals
M.__init = __init

M.MIX = 'MIX'
M.REPLACE = 'REPLACE'

def DEFAULT(win, current):
  if GLOBALS.basebg:
    ## basebg was previously used as a semi-tint mechanism used in place of the Normalbg
    ## this doesn't exactly reproduce the same colors but gets close
    return {
      'fg': {
        'rgb': COLOR_UTIL.toRgb(GLOBALS.basebg),
        'intensity': 0.5,
        'type': M.MIX,
      }
    }
  else:
    return None 
  
